lib/iris/util.py

```diff
@@ -1599,3 +1599,10 @@ def demote_dim_coord_to_aux_coord(cube, name_or_coord):
     cube.remove_coord(dim_coord)
 
     cube.add_aux_coord(dim_coord, coord_dim)
+
+
+def is_dask_array(data):
+    result = False
+    if hasattr(data, 'compute'):
+        result = True
+    return result
```

Comment:
we've been round this loop a few times

at present there is concern that we may want to change the 'how do we check for this' pattern.  Hence the suggestion of a util function.
however, this is in the public API and expected to continue, which we may not want
to be continued, i think
perhaps the death of the private `iris._lazy_data` or it's offspring was declared early?